docs/src/utils/stickWhenNeeded.ts

```diff
@@ -0,0 +1,66 @@
+import $ from 'jquery'
+
+export const stickWhenNeeded = (id: string) => {
+  // Find more elegant solution
+  var $window = $(window)
+  var lastScrollTop = $window.scrollTop()
+  var wasScrollingDown = true
+  var $sidebar = $(id)
+  if ($sidebar.length > 0) {
+    var initialSidebarTop = $sidebar.position().top
+
+    $window.scroll(() => {
+      var windowHeight = $window.height()
+      var sidebarHeight = $sidebar.outerHeight()
+
+      var scrollTop = $window.scrollTop()
+      var scrollBottom = scrollTop + windowHeight
+
+      var sidebarTop = $sidebar.position().top
+      var sidebarBottom = sidebarTop + sidebarHeight
+
+      var heightDelta = Math.abs(windowHeight - sidebarHeight)
+      var scrollDelta = lastScrollTop - scrollTop
+
+      var isScrollingDown = scrollTop > lastScrollTop
+      var isWindowLarger = windowHeight > sidebarHeight
+
+      // Hack: gradually increase the height of the container as we scroll down
+      $sidebar.css('height', `calc(100vh - ${Math.max(0, sidebarTop - scrollTop)}px)`)
```

Comment:
This is bad but meh, most of it's gonna go away at some point anyway. This sets the proper height of the TOC container relative to the scroll position. It basically goes from 150px (size of the header) to 0px when we reach the sticky position.